core/sql/executor/cluster.cpp

```diff
@@ -2395,7 +2395,7 @@ NABoolean Cluster::checkAndSplit(ExeErrorCode * rc)
 				         rc);
   
   if ( !next_ || *rc ) {
```

Comment:
Oops.  Please remove the || *rc,

Reply:
I cann't understand.
Why remove the || *rc,
I think there are two type results：
1、 First, if there are no memory, next_ will get NULL value.
2、 Second, there are some wrong happened in Cluster::Cluster
In the second case, next_ will get a value, but *rc will is not EXE_OK.
If we remove the || *rc,  this check will pass, but there was an error happened.

My understanding is wrong?